src/filemap.rs

```diff
@@ -85,21 +85,38 @@ pub fn write_file(text: &StringBuffer,
         }
     }
 
+    fn diff(text: &StringBuffer,
+            filename: &str,
+            config: &Config)
+            -> Result<Vec<Mismatch>, io::Error> {
+        let mut f = try!(File::open(filename));
+        let mut ori_text = String::new();
+        try!(f.read_to_string(&mut ori_text));
+        let mut v = Vec::new();
+        try!(write_system_newlines(&mut v, text, config));
```

Comment:
Yes I also thought it was rather wasteful but I missed the fact that I could do it the way I am doing it now. 
Sorry for the delay, switched to Arch and I have spent the week configuring it... lots of fun.